homeassistant/components/camera/nest.py

```diff
@@ -0,0 +1,93 @@
+"""
+Support for Nest Cameras.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.nest/
+"""
+
+import logging
+from datetime import timedelta
+import requests
+# import voluptuous as vol
+from homeassistant.components.camera import (PLATFORM_SCHEMA, Camera)
+import homeassistant.components.nest as nest
+from homeassistant.util import Throttle
+
+
+DEPENDENCIES = ['nest']
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({})
+# TODO be NestAware subscription aware, (10/min subscribed, 2/min otherwise)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=6)
+#MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+SIMULATOR_SNAPSHOT_URL = 'https://developer.nest.com/simulator/api/v1/nest/devices/camera/snapshot'
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup a Nest Cam."""
+    camera_devices = hass.data[nest.DATA_NEST].camera_devices()
+    cameras = [NestCamera(structure, device)
+               for structure, device in camera_devices]
+    add_devices(cameras)
+
+
+class NestCamera(Camera):
+    """Representation of a Nest Camera."""
+
+    def __init__(self, structure, device):
+        """Initialize a Nest Camera."""
+        super().__init__()
+        self.structure = structure
+        self.device = device
+
+        # data attributes
+        self._location = None
+        self._name = None
+        self._is_online = None
+        self._is_streaming = None
+        self._is_video_history_enabled = False
+
+
+    # FIXME ends up with double name, ie Hallway(Hallway (E5C0))... maybe that's just the simulator?
+    # FIXME duplication with climate/nest
+    @property
+    def name(self):
+        """Return the name of the nest, if any."""
+        self.update()
```

Comment:
If I don't include this line, then I end up with an unnamed device:

![](https://cl.ly/2c2d3Z0j2j1N/Image%202016-11-12%20at%2011.14.31%20PM.png)

Even after the image loads once, it's still unnamed:

![](https://cl.ly/3D1s1q0x1G19/Image%202016-11-12%20at%2011.15.20%20PM.png)